### PR TITLE
added sentry

### PR DESCRIPTION
**Pull request by Palabola**
## Summary by CodeRabbit

- **New Features**
  - Integrated Sentry for improved error tracking and performance monitoring in the application.
  - Added a new script for Sentry sourcemaps to enhance error tracking capabilities.

- **Bug Fixes**
  - Enhanced error reporting capabilities, contributing to better monitoring and debugging of application issues.

- **Chores**
  - Updated project dependencies to include Sentry and reorganized existing dependencies for better management.
  - Enhanced build configuration by adding source maps for easier debugging.

### CODE REVIEW

**Comment by daroczig** [comment]
currently this fails when the env var is not defined:

```
TypeError: Cannot read properties of undefined (reading 'NG_APP_SENTRY_DSN')
```

this should be optional. when passing an empty `DSN` to sentry at init, it will not actually run, so IMO just the env var lookup needs to be fixed, and potentially export the sample rate as well.

TODO update README with the new env var(s)

File: src/app/pages/server-details/server-details.component.ts [hunk 1]
```diff
@@ -408,7 +408,7 @@ export class ServerDetailsComponent implements OnInit, OnDestroy {
           }
         }
       }).catch((error) => {
-        console.error('Failed to load server data:', error);
+        this.analytics.SentryException(error, {tags: { location: this.constructor.name, function: 'getServers' }});
```
Comment (daroczig) [inline src/app/pages/server-details/server-details.component.ts:411]: I don't think we want to get a sentry exception for all 404 requests

File: src/main.server.ts [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+const SENTRY_ENVIRONMENT = import.meta?.env?.NG_APP_SENTRY_ENVIRONMENT || 'development';
+// @ts-expect-error i want this to compile, but it doesn't
+const SENTRY_RELEASE = import.meta?.env?.NG_APP_SENTRY_RELEASE || undefined;
+
+const BACKEND_BASE_URI = import.meta.env['NG_APP_BACKEND_BASE_URI'];
+
+if(SENTRY_DSN && SENTRY_DSN !== '') {
+  Sentry.init({
+    dsn: SENTRY_DSN,
+    environment: SENTRY_ENVIRONMENT + 'server',
```
Comment (daroczig) [inline src/main.server.ts:22]: I think this is still not OK :(

The environment is dev/staging/prod etc, which is very different from differentiating between client-side rendering VS server-side rendering. Let's use a custom context or tag for SSR?

Comment (daroczig) [reply src/main.server.ts:22]: probably tagging so that we can search, e.g. `rendered_at`: `client` or `server`.

**Comment by daroczig** [comment]
hey @coderabbitai, this PR is making the page loads super slow compared to the `main` branch. from the introduced sentry, it seems that the SSR part is fast, but there is a super long wait on the client side, and I am not sure why. some context: loading our ToS took 700ms, now it renders in SSR withing a few hundred ms, but then the client waits 6 seconds, without much details in sentry. do you know what's happening?

**Comment by daroczig** [comment]
@coderabbitai, this PR makes the page load super slow compared to the `main` branch: looking at sentry, it seems that the SSR part is still fast, but there is a super long wait on the client side, and I am not sure why.

some extra context: loading our ToS in the `main` branch takes 700ms, now it renders in SSR within a few hundred ms recorded as a `navigation` event in sentry, but then the client waits 6 seconds without much details in sentry as a `pageload` event.

do you know what might be happening?

**Comment by daroczig** [comment]
@coderabbitai, this PR makes the page load super slow compared to the `main` branch: looking at sentry, it seems that the SSR part is still fast, but there is a super long wait on the client side, and I am not sure why.

some extra context: loading our ToS in the `main` branch takes 700ms, now it renders in SSR within a few hundred ms recorded as a `navigation` event in sentry, but then the client waits 6 seconds without much details in sentry as a `pageload` event.

do you know what might be happening?

File: src/app/pages/server-details/server-details.component.ts (hunk 1 above)
Comment (Palabola) [reply src/app/pages/server-details/server-details.component.ts:411]: Definitely not.

File: src/main.server.ts (hunk 2 above)
Comment (Palabola) [reply src/main.server.ts]: I would be better to use diffierent DSNs for server and client side data anyways. This is just leftover code.

**Review by daroczig — APPROVED** [review]

**Pull request merged**